tracer/src/Datadog.Trace/Ci/Agent/Payloads/CICodeCoveragePayload.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="CICodeCoveragePayload.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Text;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Ci.Coverage.Models;
+using Datadog.Trace.Vendors.MessagePack;
+
+namespace Datadog.Trace.Ci.Agent.Payloads
+{
+    internal class CICodeCoveragePayload : MultipartPayload
+    {
+        public CICodeCoveragePayload(int maxItemsPerPayload = DefaultMaxItemsPerPayload, int maxBytesPerPayload = DefaultMaxBytesPerPayload, IFormatterResolver formatterResolver = null)
+            : base(maxItemsPerPayload, maxBytesPerPayload, formatterResolver)
+        {
+            var agentlessUrl = CIVisibility.Settings.AgentlessUrl;
+            if (!string.IsNullOrWhiteSpace(agentlessUrl))
+            {
+                var builder = new UriBuilder(agentlessUrl);
+                builder.Path = "api/v2/citestcov";
+                Url = builder.Uri;
+            }
+            else
+            {
+                var builder = new UriBuilder("https://datadog.host.com/api/v2/citestcov");
+                builder.Host = "event-platform-intake." + CIVisibility.Settings.Site;
+                Url = builder.Uri;
+            }
+        }
+
+        public override Uri Url { get; }
+
+        public override bool CanProcessEvent(IEvent @event)
+        {
+            return @event is CoveragePayload;
+        }
+
+        protected override MultipartFormItem CreateMultipartFormItem(ArraySegment<byte> eventInBytes)
+        {
+            var index = Count + 1;
+            return new MultipartFormItem($"coverage{index}", MimeTypes.MsgPack, $"filecoverage{index}.msgpack", eventInBytes);
+        }
+
+        public override MultipartFormItem[] ToArray()
+        {
+            // This is a current limitation in the backend side
+            // an "event" item must be included in each payload.
+            // So here we are adding a dummy one.
+            AddMultipartFormItem(
```

Comment:
nit: This would add two items if `ToArray` got called twice in a row without calling `Clear` in-between.

Reply:
yeah, you're right, I've changed this in [442a4fc](https://github.com/DataDog/dd-trace-dotnet/pull/2983/commits/442a4fc8d5b72a54fd063255b33d2d2a6e00da2b)

Reply:
Personal preference, but I would just call `Clear` from the constructor now instead of duplicating `AddMultipartFormItem`.

Reply:
Agreed, and if you do, maybe calling it `Reset()` would be a bit better?

Reply:
Changed in [08095f3](https://github.com/DataDog/dd-trace-dotnet/pull/2983/commits/08095f3ce582bf4337ed4e5e4187a9c50e531817)